### PR TITLE
Bug fix for prof_active switch

### PR DESCRIPTION
**Pull request by umanwizard**
The bug is subtle but critical: if application performs the following
three actions in sequence: (a) turn `prof_active` off, (b) make at
least one allocation that triggers the malloc slow path via the
`if (unlikely(bytes_until_sample < 0))` path, and (c) turn
`prof_active` back on, then the application would never get another
sample (until a very very long time later).

The fix is to properly reset `bytes_until_sample` rather than
throwing it all the way to `SSIZE_MAX`.

A side minor change is to call `prof_active_get_unlocked()` rather
than directly grabbing the `prof_active` variable - it is the very
reason why we defined the `prof_active_get_unlocked()` function.

### CODE REVIEW

**Review by BusyJay — DISMISSED** [review]
Thanks! But you need to get signoff check pass.

**Comment by umanwizard** [comment]
I can’t, apparently. The DCO check is checking that the signed-off-by field matches the commit author, which is not me.

This is just a cherry-pick of an upstream jemalloc commit, so I’m not sure I should change the commit author to myself…

**Comment by BusyJay** [comment]
> The DCO check is checking that the signed-off-by field matches the commit author,

It will check either author or committer. See https://github.com/probot/dco/blob/018d78c4244e3d92e34a0f4ed95e5b17b351f6b8/lib/dco.js#L52-L56.

> This is just a cherry-pick of an upstream jemalloc commit

You should not and don't need to. When cherry picking a patch, git will make you as the committer.

**Comment by umanwizard** [comment]
Thanks for the help, @BusyJay . I have fixed the signoff message appropriately.

**Review by BusyJay — APPROVED** [review]
Thanks, @sticnarf PTAL

**Review by sticnarf — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**
